src/Orleans/Core/Grain.cs

```diff
@@ -278,9 +274,12 @@ internal string CaptureRuntimeEnvironment()
     /// Base class for a Grain with declared persistent state.
     /// </summary>
     /// <typeparam name="TGrainState">The class of the persistent state object</typeparam>
-    public class Grain<TGrainState> : Grain
-        where TGrainState : GrainState
+    public class Grain<TGrainState> : Grain, IStatefulGrain
+        where TGrainState : class
     {
+        private readonly GrainState<TGrainState> grainState;
+
+        private IStorage storage { get; set; }
```

Comment:
should this be a variable instead of an auto prop? We usually capitalize property names.